python/hsfs/engine/python.py

```diff
@@ -466,7 +473,15 @@ def _split_df(self, df, splits):
         random.shuffle(groups)
         df[split_column] = groups
         for i, item in enumerate(items):
-            result_dfs[item[0]] = df[df[split_column] == i].drop(split_column, axis=1)
+            split_df = df[df[split_column] == i].drop(split_column, axis=1)
+            if item == training_dataset_obj.train_split:
```

Comment:
May be move the transformation part out of this method? Now this method not only do "_split_df".

Reply:
Not sure if I follow: here  `populate_builtin_transformation_functions` needs to know which split to use and `_apply_transformation_function` is applied one by one to splits which is different if there are no splits

Reply:
I mean keep this `split_df` method unchange but `populate_builtin_transformation_functions` and `_apply_transformation_function` in `get_training_data`.

Reply:
well I need to get `train_split`. under `if training_dataset_obj.splits` I just know that there is split and then I have to extract `train_split` stats. thats why I put this under a loop in `_split_df` to get that info .

Reply:
I think the code is almost the same, the difference is just where to put it. I think it is clearer to put it outside of this method because reader may think that transformation is not applied when there is a split.. But I am ok with current implementation as well, but please rename this method name so that it is clear that it also apply transformation.

Reply:
done